### ISSUE TITLE
Cannot sign in to GitHub.com, (Version 2.8.3)

### ISSUE THREAD

**Issue by benoitberanger** [shiftkey/desktop#524]
### Describe the bug

It seems that I cannot sign / link Desktop to github.com

### Version & OS

Version 2.8.3-linux1 (x64) - June 2nd 2021

Distributor ID:	LinuxMint
Description:	Linux Mint 19 Tara
Release:	19
Codename:	tara

XFCE desktop

### Steps to reproduce the behavior

1. click *File* > *Clone repository*
2. popup : *Clone repo* with 3 option, select GitHub.com, click *Sign in*
3. popup : *Sign in using your browser*, click *Continue with browser*
4. my browser Firefox opens a new tab with a popup *choose an app to open x-github-desktop-dev-auth-link*, select *GitHub Desktop*, click *Open link*
5. On Desktop, the last popup *Sign in using your browse* disappears, on browser, the *choose app* popup disappears
Nothing...

I can repeat this process 1 to 5, no difference.

### Expected behavior

I expected a new page on GitHub.com that would confirm I authorize Desktop to have access to my account. But nothing appends.

### Actual behavior

It seems there is no authorization from GitHub.com that Desktop have access credentials

### Logs

```
2021-06-10T16:03:56.270Z - info: [ui] [AppStore] loading 11 repositories from store
2021-06-10T16:03:56.596Z - info: [ui] launching: 2.8.3-linux1 (Linux 4.15.0-142-lowlatency)
2021-06-10T16:03:56.596Z - info: [ui] execPath: '/usr/lib/github-desktop/github-desktop'
2021-06-10T16:05:04.529Z - info: [ui] [SignInStore] initializing OAuth flow
2021-06-10T16:05:04.534Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=3a723b10ac5575cc5bb9&scope=repo%20user%20workflow&state=1730d935-c7a6-40ad-92db-97ac087fb5ff
2021-06-10T16:05:15.954Z - info: [main] Received possible protocol arguments: 2
2021-06-10T16:05:15.954Z - info: [main] Processing protocol url
2021-06-10T16:05:15.961Z - info: [ui] [Dispatcher] requesting authenticated user
2021-06-10T16:05:16.559Z - info: [ui] [SignInStore] account resolved
2021-06-10T16:05:16.560Z - info: [ui] [AppStore] adding account benoitberanger (Benoît Béranger) to store

```
### Additional context

I could not install with `apt-get`, it seems this is common problem of limited bandwidth from the .deb provider
So I directly download the .deb in the *Release* section, and installed with `dpkg -i`

**Comment by shiftkey** [comment]
@benoitberanger thanks for the report. What's weird here is that the last step seems to work - receiving the token and storing it in the OS credential store (using `libsecret`):

```
2021-06-10T16:05:16.560Z - info: [ui] [AppStore] adding account benoitberanger (Benoît Béranger) to store
```

I see you're using XFCE, but perhaps there's no `libsecret`-compatible service running to store the credential. We use `gnome-keyring` but if that's not present (or not working as expected) that could be a source of the issue you're seeing...

**Comment by benoitberanger** [comment]
@shiftkey Thank you for your time.

`libsecret` is present in my system : 

```
benoit.beranger@icm-cenir-le33: ~ $ locate libsecret
/usr/lib/x86_64-linux-gnu/libsecret-1.so.0
/usr/lib/x86_64-linux-gnu/libsecret-1.so.0.0.0
/usr/lib/x86_64-linux-gnu/samba/libsecrets3.so.0
/usr/lib/x86_64-linux-gnu/vlc/plugins/keystore/libsecret_plugin.so
/usr/share/doc/libsecret-1-0
/usr/share/doc/libsecret-common
/usr/share/doc/git/contrib/credential/libsecret
/usr/share/doc/git/contrib/credential/libsecret/Makefile
/usr/share/doc/git/contrib/credential/libsecret/git-credential-libsecret.c
/usr/share/doc/libsecret-1-0/changelog.Debian.gz
/usr/share/doc/libsecret-1-0/copyright
/usr/share/doc/libsecret-common/changelog.Debian.gz
/usr/share/doc/libsecret-common/copyright
/usr/share/locale-langpack/en_AU/LC_MESSAGES/libsecret.mo
/usr/share/locale-langpack/en_CA/LC_MESSAGES/libsecret.mo
/usr/share/locale-langpack/en_GB/LC_MESSAGES/libsecret.mo
/var/lib/dpkg/info/libsecret-1-0:amd64.list
/var/lib/dpkg/info/libsecret-1-0:amd64.md5sums
/var/lib/dpkg/info/libsecret-1-0:amd64.shlibs
/var/lib/dpkg/info/libsecret-1-0:amd64.symbols
/var/lib/dpkg/info/libsecret-1-0:amd64.triggers
/var/lib/dpkg/info/libsecret-common.list
/var/lib/dpkg/info/libsecret-common.md5sums
```

It seems to be the case for `gnome-keyring` as well :

```
benoit.beranger@icm-cenir-le33: ~ $ locate gnome-keyring
/etc/xdg/autostart/gnome-keyring-pkcs11.desktop
/etc/xdg/autostart/gnome-keyring-secrets.desktop
/etc/xdg/autostart/gnome-keyring-ssh.desktop
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.1.gz
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.2.gz
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.3.gz
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.4.gz
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.5.gz
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.6.gz
/home/benoit.beranger/.cache/upstart/gnome-keyring-ssh.log.7.gz
/usr/bin/gnome-keyring
/usr/bin/gnome-keyring-3
/usr/bin/gnome-keyring-daemon
/usr/lib/python3/dist-packages/orca/scripts/apps/gnome-keyring-ask
/usr/lib/python3/dist-packages/orca/scripts/apps/gnome-keyring-ask/__init__.py
/usr/lib/python3/dist-packages/orca/scripts/apps/gnome-keyring-ask/__pycache__
/usr/lib/python3/dist-packages/orca/scripts/apps/gnome-keyring-ask/script.py
/usr/lib/python3/dist-packages/orca/scripts/apps/gnome-keyring-ask/__pycache__/__init__.cpython-36.pyc
/usr/lib/python3/dist-packages/orca/scripts/apps/gnome-keyring-ask/__pycache__/script.cpython-36.pyc
/usr/lib/systemd/user/gnome-keyring-ssh.service
/usr/lib/systemd/user/gnome-keyring.service
/usr/lib/systemd/user/graphical-session-pre.target.wants/gnome-keyring-ssh.service
/usr/lib/systemd/user/graphical-session-pre.target.wants/gnome-keyring.service
/usr/lib/x86_64-linux-gnu/gnome-keyring
/usr/lib/x86_64-linux-gnu/gnome-keyring/devel
/usr/lib/x86_64-linux-gnu/gnome-keyring/devel/gkm-gnome2-store-standalone.so
/usr/lib/x86_64-linux-gnu/gnome-keyring/devel/gkm-secret-store-standalone.so
/usr/lib/x86_64-linux-gnu/gnome-keyring/devel/gkm-ssh-store-standalone.so
/usr/lib/x86_64-linux-gnu/gnome-keyring/devel/gkm-xdg-store-standalone.so
/usr/lib/x86_64-linux-gnu/pkcs11/gnome-keyring-pkcs11.so
/usr/share/doc/gnome-keyring
/usr/share/doc/gnome-keyring-pkcs11
/usr/share/doc/libp11-kit-gnome-keyring
/usr/share/doc/libpam-gnome-keyring
/usr/share/doc/git/contrib/credential/gnome-keyring
/usr/share/doc/git/contrib/credential/gnome-keyring/Makefile
/usr/share/doc/git/contrib/credential/gnome-keyring/git-credential-gnome-keyring.c
/usr/share/doc/gnome-keyring/AUTHORS
/usr/share/doc/gnome-keyring/NEWS.gz
/usr/share/doc/gnome-keyring/README
/usr/share/doc/gnome-keyring/README.Debian
/usr/share/doc/gnome-keyring/changelog.Debian.gz
/usr/share/doc/gnome-keyring/copyright
/usr/share/doc/gnome-keyring-pkcs11/changelog.Debian.gz
/usr/share/doc/gnome-keyring-pkcs11/copyright
/usr/share/doc/libp11-kit-gnome-keyring/changelog.Debian.gz
/usr/share/doc/libp11-kit-gnome-keyring/copyright
/usr/share/doc/libpam-gnome-keyring/README.Debian
/usr/share/doc/libpam-gnome-keyring/changelog.Debian.gz
/usr/share/doc/libpam-gnome-keyring/copyright
/usr/share/locale-langpack/en@shaw/LC_MESSAGES/gnome-keyring.mo
/usr/share/locale-langpack/en_AU/LC_MESSAGES/gnome-keyring.mo
/usr/share/locale-langpack/en_CA/LC_MESSAGES/gnome-keyring.mo
/usr/share/locale-langpack/en_GB/LC_MESSAGES/gnome-keyring.mo
/usr/share/man/man1/gnome-keyring-3.1.gz
/usr/share/man/man1/gnome-keyring-daemon.1.gz
/usr/share/man/man1/gnome-keyring.1.gz
/usr/share/pam-configs/gnome-keyring
/var/lib/dpkg/info/gnome-keyring-pkcs11:amd64.list
/var/lib/dpkg/info/gnome-keyring-pkcs11:amd64.md5sums
/var/lib/dpkg/info/gnome-keyring.conffiles
/var/lib/dpkg/info/gnome-keyring.list
/var/lib/dpkg/info/gnome-keyring.md5sums
/var/lib/dpkg/info/gnome-keyring.postinst
/var/lib/dpkg/info/gnome-keyring.postrm
/var/lib/dpkg/info/gnome-keyring.preinst
/var/lib/dpkg/info/gnome-keyring.prerm
/var/lib/dpkg/info/libp11-kit-gnome-keyring:amd64.list
/var/lib/dpkg/info/libp11-kit-gnome-keyring:amd64.md5sums
/var/lib/dpkg/info/libpam-gnome-keyring:amd64.list
/var/lib/dpkg/info/libpam-gnome-keyring:amd64.md5sums
/var/lib/dpkg/info/libpam-gnome-keyring:amd64.postinst
/var/lib/dpkg/info/libpam-gnome-keyring:amd64.prerm
```

Shall I do / check something else ?
Let me kow If I can do anything to solve the problem.

**Comment by shiftkey** [comment]
As I can't see any issues in the logs about adding the token to the keyring, I'd open **File** | **Options** and check the **Accounts** tab:

![](https://user-images.githubusercontent.com/359239/122235986-819d8d80-ce94-11eb-8e26-9b2f5036240d.png)

If you see your account there, and can fetch repositories through the app, I think everything is working and there was just something weird about the browser callback.

> I expected a new page on GitHub.com that would confirm I authorize Desktop to have access to my account.

I think some in-app confirmation here that you've completed the signup flow would be great, but I don't have the bandwidth for adding features like this these days...

**Comment by benoitberanger** [comment]
I do not see my GitHub account there : 
![image](https://user-images.githubusercontent.com/16976839/122238192-3a2cf680-cec0-11eb-884a-eeaaa8a5f4c5.png)

When I "sign in" to GitHub, a popup "sign in using your browser", click "continue with browser".
Then a Firefox popup :
![image](https://user-images.githubusercontent.com/16976839/122238533-7eb89200-cec0-11eb-94e1-dd91932c8f69.png)

Click "open link", and a new window appears : 
![image](https://user-images.githubusercontent.com/16976839/122239018-e2db5600-cec0-11eb-9531-0238859fd0bd.png)

I tried my PC user password (which has sudo rights), doesnt work.
I tried my Github.com password, doesnt work.
When I click "cancel", then a popup in Desktop : 
![image](https://user-images.githubusercontent.com/16976839/122239389-2df56900-cec1-11eb-9462-d363ff7507dd.png)

When I click "close" I see a few subliminal popups that open and close immediately.

**Comment by shiftkey** [comment]
Ok, that's helpful context - seems like something isn't setup right with `gnome-keyring` so at least we know the app is able to interact with `libsecret`.

**Comment by shiftkey** [comment]
When I had this happen before, I deleted `~/.local/share/keyrings/login.keyring` and restarted so a fresh keyring file would be created. The internet isn't exactly clear on other options, unless you can remember the original password that might have been used to create this keychain...

**Comment by benoitberanger** [comment]
This is my company desktop, we use company wide login system (don't have the name in mind). Our passwords changes periodically. So I might have created a "keyring" a very long time ago and obviously I don't remember the old password I used at that moment. I probably used one of the periodic password for the keyring because.

I renamed the `login.keyring` to `login.keyring.backup_2021-06-16` to try.
After the Firefox "open link" popup I created a **new** keyring with a fresh password that I will remember that is not one the periodic ones.
Now I'm logged in in Desktop : 
![image](https://user-images.githubusercontent.com/16976839/122244595-4b2c3680-cec5-11eb-833b-ebcf854686bf.png)

This keyring process is still obscure to me. But now access to my GitHub.com repos from Desktop, and  know how to rollback with the backup file.

Thank you for your help @shiftkey.
